calico-vpp-agent/cmd/calico_vpp_dataplane.go

```diff
@@ -151,23 +151,19 @@ func main() {
 	routingServer.SetBGPConf(bgpConf)
 	serviceServer.SetBGPConf(bgpConf)
 
+	watchDog := NewWatchDog(log.WithFields(logrus.Fields{"component": "watchDog"}))
+	Go(watchDog.watch)
 	Go(policyServer.ServePolicy)
-	log.Infof("Waiting for our node's BGP spec...")
 	felixConfig := policyServer.WaitForFelixConfig()
+	felixConfigReceived = true
```

Comment:
Instead of relying on variables, we could do something like 
````go
watchDog.Wait(felixConfig, "Waiting for FelixConfig to be provided by the calico pod")
// with 
type WatchDogWatchable interface (
   func GetWaitChan() <-struct{}
)
````
That will allow disconnections between the variables and the `WaitForFelixConfig()` calls if refactoring happen at some point. Also we can do a select on this chan but also on the tomb (to return in case something fails)

Reply:
addressed in #505